### PR TITLE
[Perf] Optimize MTP execution by reordering state update operation

### PR DESCRIPTION
**Pull request by SlightwindSec**
## Summary
- Move `_update_states_after_model_execute` call from after main model sampling to after draft model execution
- This reordering reduces pipeline bubbles between main model and draft model execution
- No accuracy impact - the state update operation is independent of draft token proposal

## Performance Impact
Reduces idle time between main model and draft model execution stages, improving overall MTP (Multi-Token Prediction) performance.
- vLLM version: v0.15.0
- vLLM main: https://github.com/vllm-project/vllm/commit/83b47f67b1dfad505606070ae4d9f83e50ad4ebd

### CODE REVIEW

File: vllm_ascend/worker/model_runner_v1.py [hunk 1]
```diff
@@ -1447,7 +1449,12 @@ def sample_tokens(
             sampler_output = self._sample(logits, spec_decode_metadata)
 
         if self.need_accepted_tokens:
-            self._update_states_after_model_execute(sampler_output.sampled_token_ids, scheduler_output)
+            if self.state_update_stream is None:
```
Comment (weijinqian0) [inline vllm_ascend/worker/model_runner_v1.py]: this may use global_stream in utils.py when there is not conflict.
